src/app/modules/form/admin-form/admin-form.payments.controller.ts

```diff
@@ -309,25 +310,62 @@ export const _handleUpdatePayments: ControllerHandler<
   )
 }
 
+export const handleUpdatePaymentsForTest = _handleUpdatePayments
+
+const updatePaymentsValidator = celebrate({
+  [Segments.BODY]: {
+    enabled: Joi.boolean().required(),
+    payment_type: Joi.string()
+      .allow(...Object.values(PaymentType))
+      .required(),
+    amount_cents: Joi.when('enabled', {
+      is: Joi.equal(true),
+      then: Joi.when('payment_type', {
+        is: Joi.equal(PaymentType.Fixed),
+        then: Joi.number().integer().positive().required(),
+        otherwise: Joi.number(),
```

Comment:
All `amount_cents`, `min_amount`, `max_amount` should be integers. 

The reason I'm not validating it, aside from ensuring it as `typeof number`, is that the value is not read/used if it is not `PaymentType.Fixed`.

Reply:
Ah I see what you mean. The outer `otherwise` checks for `integer()` but inner `otherwise` doesn't. This can cause conflicts when the user switches from `enable` to `disable`.

Reply:
7b51943d3